### PR TITLE
Nuxt integration

### PR DESCRIPTION
**Pull request by Bijit-Mondal**
## Summary of change

Added Nuxt JS to the full-stack framework list, using the prebuilt UI.

## Checklist for important updates

- [x] Changelog has been updated
- [x] Changes to the version if 
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] If added a new recipe, I also modified types to include the new recipe in `Recipe` and `allRecipes`
-   [x] If added a new frontend, I also modified types to include the new frontend in `SupportedFrontends` and `allFrontends` if required
-   [x] If added a new backend, I also modified types to include the new backend in `SupportedBackends` and `allBackends` if required

### CODE REVIEW

File: boilerplate/fullstack/nuxt/pages/auth/[...slug].vue [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        script.type = "text/javascript";
+        script.src = src;
+        script.id = "supertokens-script";
+        script.onload = () => {
+            initSuperTokensUI();
+        };
+        document.body.appendChild(script);
+    };
+
+    loadScript("https://cdn.jsdelivr.net/gh/supertokens/prebuiltui@v0.47.1/build/static/js/main.ba50d5ee.js");
```
Comment (rishabhpoddar) [inline boilerplate/fullstack/nuxt/pages/auth/[...slug].vue:16]:
```suggestion
    loadScript("${jsdeliveryprebuiltuiurl}");
```

Comment (rishabhpoddar) [reply boilerplate/fullstack/nuxt/pages/auth/[...slug].vue:16]: All occurrences of the js bundle should be changed to the above string. The CLI replaces it with the latest bundle url during the creation of the app

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
```
Comment (rishabhpoddar) [inline CHANGELOG.md:9]: Could you take a loom of testing out the code and generating a nuxt js app?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: boilerplate/fullstack/nuxt/server/api/auth.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { handleAuthAPIRequest } from "../superTokensHelper";
+import { ensureSuperTokensInit } from "../backend";
+
+ensureSuperTokensInit();
+
```
[the remark this answers is not in the document]
Comment (Bijit-Mondal) [reply boilerplate/fullstack/nuxt/server/api/auth.ts:5]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (rishabhpoddar) [reply CHANGELOG.md:9]: Please also include a call to the sessioninfo endpoint with the session tokens in the demo. Thanks

Comment (Bijit-Mondal) [reply CHANGELOG.md:9]: Here it is
https://jam.dev/c/a3625633-1562-4d7d-a5b5-895492caa27e

Comment (rishabhpoddar) [reply CHANGELOG.md:9]: Im getting the following error when i try and call the sign in API:
```
Unexpected token o in JSON at position 1

  at JSON.parse (<anonymous>)
  at vr.json (node_modules/node-fetch-native/dist/node.mjs:17:2658)
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
  at Object.getJSONBody (server/superTokensHelper.ts:121:20)
  at async PreParsedRequest.BaseRequest.getJSONBody (node_modules/supertokens-node/lib/build/framework/request.js:40:39)
  at async Object.signInAPI [as default] (node_modules/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:33:10)
  at async Recipe.handleAPIRequest (node_modules/supertokens-node/lib/build/recipe/emailpassword/recipe.js:103:24)
  at async SuperTokens.middleware (node_modules/supertokens-node/lib/build/supertokens.js:256:38)
  at async node_modules/supertokens-node/lib/build/framework/custom/framework.js:133:28
  at handleCall (server/superTokensHelper.ts:58:36)
```

Comment (Bijit-Mondal) [reply CHANGELOG.md:9]: Would you please elaborate, I am not getting any error. 
make sure you have 
- [ ] generated nuxt js app using `npx . --branch=feat/nuxt-prebuilt`
- [ ] `cd <directory>` `npm install` and `npm run dev`. Make sure it's open on port `3000`.

for me signin is working fine. 

If this doesn't resolve the issue, please share a loom.

Comment (Bijit-Mondal) [reply CHANGELOG.md:9]: is this issue still exist?

Comment (rishabhpoddar) [reply CHANGELOG.md:9]: @deepjyoti30-st from our team will try it out and get back to you.

Comment (Bijit-Mondal) [reply CHANGELOG.md:9]: Hi @deepjyoti30-st please let me know if it is working

File: lib/ts/config.ts [hunk 4]
```diff
@@ -202,6 +202,32 @@ export async function getFrontendOptions({ manager }: UserFlags): Promise<Questi
                 setup: [`${manager} install`],
             },
         },
```
Comment (rishabhpoddar) [inline lib/ts/config.ts:204]: We also need to modify the index.ts near line number 131 to say that Nuxt.js app doesn't support multi tenancy

**Review by rishabhpoddar — DISMISSED** [review]

File: lib/ts/config.ts (hunk 4 above)
Comment (Bijit-Mondal) [reply lib/ts/config.ts:204]: Done

File: boilerplate/fullstack/nuxt/server/utils/convertToRequest.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { H3Event } from "h3";
+import { appInfo } from "~/config/appInfo";
+
+interface ExtendedRequestInit extends RequestInit {
+    duplex?: string;
+}
+
+export async function convertToRequest(event: H3Event): Promise<Request> {
+    const url = new URL(`${appInfo.apiDomain}${event._path}`);
```
Comment (deepjyoti30-st) [inline boilerplate/fullstack/nuxt/server/utils/convertToRequest.ts:9]: Can we safely assume that `event._path` will always have a prefix `/`?

**Review by deepjyoti30-st — CHANGES_REQUESTED** [review]
I have tested it and it's working fine functionality wise. Requested some improvements besides.

File: boilerplate/fullstack/nuxt/pages/index.vue [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    <div class="body">
+        <h1>Hello</h1>
+
+        <div v-if="session">
+            <div>
+                <span>UserId:</span>
+                <h3>{{ userId }}</h3>
+
+                <button @click="onLogout">Sign Out</button>
+            </div>
```
[the remark this answers is not in the document]
Comment (Bijit-Mondal) [reply boilerplate/fullstack/nuxt/pages/index.vue:11]: That can be done, but as you say Astro and all other React and it's meta framework have the ui to call `/sessioninfo` from frontend in boilerplate, but all other framework(like sveltekit,angular,vue) doesn't have that. 

To maintain uniformity we need to change in all non React framework, that might be good to do in another PR?

File: boilerplate/fullstack/nuxt/server/utils/convertToRequest.ts (hunk 5 above)
Comment (Bijit-Mondal) [reply boilerplate/fullstack/nuxt/server/utils/convertToRequest.ts:9]: Yes, the prefix `/` will always be present, as mentioned in the documentation: https://h3.unjs.io/guide/event#eventpath

File: boilerplate/fullstack/nuxt/pages/index.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (Bijit-Mondal) [reply boilerplate/fullstack/nuxt/pages/index.vue:11]: being said that I have plan to revamp the UI little bit for all non react framework, so if it is acceptable we might do a seperate PR for that?

[the remark this answers is not in the document]
Comment (deepjyoti30-st) [reply boilerplate/fullstack/nuxt/pages/index.vue:11]: You can create a separate PR for other frameworks that don't have it but you can update this one with the changes before I approve it.

File: boilerplate/fullstack/nuxt/server/utils/convertToRequest.ts (hunk 5 above)
Comment (deepjyoti30-st) [reply boilerplate/fullstack/nuxt/server/utils/convertToRequest.ts:9]: Thanks.

File: boilerplate/fullstack/nuxt/pages/index.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (Bijit-Mondal) [reply boilerplate/fullstack/nuxt/pages/index.vue]: Done.

**Review by deepjyoti30-st — APPROVED** [review]

**Pull request merged**